### DISCUSSION TITLE
Check true/false

### DISCUSSION THREAD

**Discussion by chengm349** [nlohmann/json#4473] [General]
"Source": [
      {
        "Name": "Bloomberg",
        "EntitlementChecked" : true,
      }
    ]

assume source is a ref to the first object of Source array. Can I use if (source.at("EntitlementChecked")) ...  ?
if "true" is "tru", should the if stmt fail?

**Comment by chengm349** [comment]
any input?

**Reply by gregmarr** [reply]
Well, it should fail at the parse if you have a malformed JSON file because you spelled `true` wrong.  What exactly are you trying to do?

**Comment by chengm349** [comment]
true or tru or other value should not affect parsing since it's just a value? Not a name? My question is that if 

"EntitlementChecked" : true,

may I code like this:

if (source.at("EntitlementChecked"))
{
 // true ,do some
} else {
// do other
}

**Reply by gregmarr** [reply]
> true or tru or other value should not affect parsing since it's just a value? Not a name?
It's not a string, it's a binary literal.  If it were `"true"` vs `"tru"` then yes, it would still parse.

```
if (source.at("EntitlementChecked"))
{
// true ,do some
} else {
// do other
}
```

`source.at()` returns a `json` object, so no, that won't work.  You would need to do `.get<bool>()` to get the boolean value.  If you want it to behave as the value is `false` if`"EntitlementChecked"` isn't there, you should do `source.value("EntitlementChecked", false)`

**Comment by chengm349** [comment]
very clear. and thanks!
